services/src/main/java/org/exoplatform/addons/gamification/listener/challenges/AnnouncementActivityGeneratorListener.java

```diff
@@ -68,12 +87,14 @@ public void onEvent(Event<AnnouncementService, Announcement> event) throws Objec
     }
   }
 
-  private ExoSocialActivity createActivity(Announcement announcement, Challenge challenge) throws ObjectNotFoundException {
+  private ExoSocialActivity createActivity(Announcement announcement, Map<String, String> params, Challenge challenge) throws ObjectNotFoundException {
+    if (params == null) {
+      params = new HashMap<>();
+    }
```

Comment:
No need to instantiate template params with an empty map if it's null

```suggestion
    if (params != null) {
      activity.setTemplateParams(params);
    }
```

Reply:
this check is added in case no templeteParams is send when it is null
so I can add extra object